README.md

```diff
@@ -1,2 +1,112 @@
+  test_dataloader = get_data_loader("/content/data/", 32, True, split='test')
+
+  model = ResNet9(num_classes=6)
+  model.to(torch.device('cuda'))
+
+  train_model(model, train_dataloader, test_dataloader,
+              num_epochs=10, learning_rate=1e-3, checkpoint_dir='checkpoints',logger = "training.log", device=torch.device('cuda'))
+```
+
+### Segmentation  
+**Adding Data for Training**  
+- Create a 'data' folder.  
+- The 'data' folder will contain three different folders named 'train', 'test', and 'eval' used for training, testing, and evaluation purposes.  
+- Each of the 'train', 'test', and 'eval' folders contain data sets of 'images' and 'masks'. 
+- Data folder structure 'content'->'data'->('train', 'test', 'eval')->('images', 'masks').
+    
+**Training the model**  
+- Run the following code to train your model and you can change the default arguments with your custom arguments  
+
+```
+import UniTrain
+from UniTrain.utils.segmentation import get_data_loader, train_model, generate_model_summary
+from UniTrain.models.segmentation import UNet
+from UniTrain.utils.segmentation import parse_folder
+import torch
+import os
+import glob
+
+
+if parse_folder(data_dir):    
+    train_path = os.path.join(data_dir, "train")
+    test_path = os.path.join(data_dir, "test")
+    
+    train_image_paths = glob.glob(os.path.join(train_path, "images", "*.jpg"))
+    train_mask_paths = glob.glob(os.path.join(train_path, "masks", "*.png"))
+    
+    test_image_paths = glob.glob(os.path.join(test_path, "images", "*.jpg"))
+    test_mask_paths = glob.glob(os.path.join(test_path, "masks", "*.png"))
+    
+    train_data_loader = get_data_loader(image_paths=train_image_paths, mask_paths=train_mask_paths, batch_size=32, shuffle=True, transform=None)
+    test_data_loader = get_data_loader(image_paths=test_image_paths, mask_paths=test_mask_paths, batch_size=32, shuffle=True, transform=None)
+
+    model = UNet(n_class=20)
+    model.to(torch.device('cuda'))
+    
+    generate_model_summary(model=model, input_size=(3, 512, 512))
+    
+    train_unet( model, train_data_loader, test_data_loader, num_epochs=10, learning_rate=1e-3, checkpoint_dir='checkpoints', logger="training.log",iou=False, device=torch.device('cuda'))
+```
+
+### DCGAN
+
+**Adding Data for Training**  
+- For the data create a folder 'data'->'realimages'->'images'-> Add your data here
+- No need to create fake images as it would be generated by the untrained generator  
+
+**Training the model ( both discriminator and generator )**  
+- Run the following code to train your model and you can change the default arguments with your custom arguments  
+
+```
+import UniTrain
+from UniTrain.utils.DCGAN import parse_folder, get_data_loader, train_model
+from UniTrain.models.DCGAN import disc, gen
+import glob
+import torch
+
+if parse_folder('data'):
+    real_image_paths = glob.glob("data/real_images/images/*.jpg")
```

Comment:
If this is how the DCGAN is implemented then we must change it, we shouldn't require users to care about external libraries such as glob for training.